service/src/main/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataService.java

```diff
@@ -0,0 +1,238 @@
+        if (isNotEmpty(respondents)) {
+            return respondents.stream()
+                .filter(Objects::nonNull)
+                .map(Element::getValue)
+                .filter(respondent -> isNotEmpty(respondent.getParty()))
+                .map(uk.gov.hmcts.reform.fpl.model.Respondent::getParty)
+                .map(this::buildRespondent)
+                .collect(toSet());
+        }
+
+        return of();
+    }
+
+    private Respondent buildRespondent(final RespondentParty respondentParty) {
+        return Respondent.builder()
+            .firstName(respondentParty.getFirstName())
+            .lastName(respondentParty.getLastName())
+            .gender(convertStringToGender(respondentParty.getGender()))
+            .address(convertAddress(respondentParty.getAddress()))
+            .relationshipToChild(respondentParty.getRelationshipToChild())
+            .dob(formatDob(respondentParty.getDateOfBirth()))
+            // TODO: 19/12/2019 verify if this should always be true ???
+            .confidential(true)
+            .build();
+    }
+
+    private Set<Child> populateChildren(final List<Element<uk.gov.hmcts.reform.fpl.model.Child>> allChildren) {
+        if (isNotEmpty(allChildren)) {
+            return allChildren.stream()
+                .filter(Objects::nonNull)
+                .map(Element::getValue)
+                .filter(child -> isNotEmpty(child.getParty()))
+                .map(uk.gov.hmcts.reform.fpl.model.Child::getParty)
+                .map(this::buildChild)
+                .collect(toSet());
+        }
+
+        return of();
+    }
+
+    private Child buildChild(final ChildParty childParty) {
+        return Child.builder()
+            .firstName(childParty.getFirstName())
+            .lastName(childParty.getLastName())
+            .gender(convertStringToGender(childParty.getGender()))
+            .dob(formatDob(childParty.getDateOfBirth()))
+            // TODO: 19/12/2019 verify if this should always be true ???
+            .isParty(true)
+            .build();
+    }
+
+    private String formatDob(final LocalDate date) {
+        return isEmpty(date) ? "" : dateFormatterService.formatLocalDateToString(date, "d-MMM-y").toUpperCase();
+    }
+
+    private String deriveApplicationType(final Orders orders) {
+        if (isEmpty(orders) && isEmpty(orders.getOrderType())) {
+            throw new RoboticsDataException("No order type(s) to derive Application Type from.");
+        }
+
+        List<OrderType> selectedOrderTypes = orders.getOrderType()
+            .stream()
+            .filter(Objects::nonNull)
+            .collect(toList());
+
+        if (selectedOrderTypes.size() > 1) {
+            return derivedApplicationTypeWithNoDuplicate(selectedOrderTypes.stream()
+                .map(this::getOrderTypeLabelValue)
+                .collect(joining(",")));
+
+        } else {
+            return getOrderTypeLabelValue(selectedOrderTypes.get(0));
+        }
```

Comment:
Don't understand why now you need to have two cases? Couldn't this be handled in one stream + map / distinct flow? 

orders.getOrderType()
              .stream()
              .filter(Objects::nonNull)
.map(this::getOrderTypeLabelValue)
 .distinct()
 .collect(joining(",")))

Wouldn't just that work?

Reply:
this will absolutely work 👍. Not sure why I went the split route 😄.  Now updated